pkg/sql/plan2/function/builtin/multi/concat.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2021 - 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multi
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/common/moerr"
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+func Concat(vectors []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	resultType := types.Type{Oid: types.T_varchar, Size: 24}
+	isAllConst := true
+
+	for i := range vectors {
+		if vectors[i].IsScalarNull() {
+			return proc.AllocScalarNullVector(resultType), nil
+		}
```

Comment:
so when concat("hello", null, "junhong") the result should be NULL? not sure this is the correct semantic

Reply:
should be NULL, same as MySQL